src/org.hdfgroup.object/hdf/object/h5/H5File.java

```diff
@@ -2053,25 +2055,27 @@ public void writeAttribute(HObject obj, Attribute attr, boolean attrExisted) thr
                     aid = H5.H5Acreate(objID, name, tid, sid, HDF5Constants.H5P_DEFAULT, HDF5Constants.H5P_DEFAULT);
                 log.trace("writeAttribute(): aid {} opened/created", aid);
 
-                // update value of the attribute
-                Object attrValue;
-                try {
-                    attrValue = attr.getAttributeData();
-                }
-                catch (Exception ex) {
-                    attrValue = null;
-                    log.trace("writeAttribute(): getAttributeData() failure:", ex);
-                }
-
-                //log.trace("writeAttribute(): attrValue={}", attrValue);
-                if (attrValue != null) {
+                if (!attr.isAttributeNULL()) {
+                    // update value of the attribute
+                    Object attrValue;
                     try {
-                        ((H5Attribute)attr).AttributeCommonIO(aid, H5File.IO_TYPE.WRITE, attrValue);
+                        attrValue = attr.getAttributeData();
                     }
                     catch (Exception ex) {
-                        log.debug("writeAttribute(): failed to write attribute: ", ex);
+                        attrValue = null;
+                        log.trace("writeAttribute(): getAttributeData() failure:", ex);
                     }
-                } // (attrValue != null)
+
+                    // log.trace("writeAttribute(): attrValue={}", attrValue);
```

Comment:
Remove the line or uncomment the line for logging.

Reply:
This is the development branch.
Debug statements will be removed or finalized in the release development branches.